src/main/java/com/android/volley/toolbox/DiskBasedCache.java

```diff
@@ -255,31 +252,28 @@ public synchronized void put(String key, Entry entry) {
     /** Removes the specified key from the cache if it exists. */
     @Override
     public synchronized void remove(String key) {
-        boolean deleted = getFileForKey(key).delete();
-        removeEntry(key);
+        boolean deleted = DiskBasedCacheUtility.getFileForKey(key, mRootDirectorySupplier).delete();
+        mTotalSize = DiskBasedCacheUtility.removeEntry(key, mTotalSize, mEntries);
         if (!deleted) {
             VolleyLog.d(
                     "Could not delete cache entry for key=%s, filename=%s",
-                    key, getFilenameForKey(key));
+                    key, DiskBasedCacheUtility.getFilenameForKey(key));
         }
     }
 
+    /** Represents a supplier for {@link File}s. */
+    public interface FileSupplier extends DiskBasedCacheUtility.FileSupplier {
+        @Override
```

Comment:
Is this necessary? I think you can just do "extends DiskBasedCacheUtility.FileSupplier { }" since you're not changing anything about the underlying method.

Reply:
Oh, I'll move the File Supplier. And you're right,  that isn't necessary,  I'll change that as well

Reply:
I had to change the name to FileSupplierParent, as otherwise, it couldn't be extended by FileSupplier

Reply:
Can you call it FileSupplier and do "public interface FileSupplier implements com.google.android.volley.toolbox.FileSupplier"?

Reply:
Yep, just tried that and it works!